ocaml/xenopsd/xc/xenops_server_xen.ml

```diff
@@ -1332,11 +1332,20 @@ module VM = struct
       @ affinity
       @ weight
     in
+    let set_generation_id platformdata =
+      let key = "generation-id" in
+      match vm.generation_id with
+      | Some genid ->
+          (key, genid) :: (List.remove_assoc key platformdata)
+      | None ->
+          platformdata
+    in
     let default k v p = if List.mem_assoc k p then p else (k, v) :: p in
```

Comment:
This is existing code but `p` does not strike me as a good name; `assoc`, `kvs` would create a stronger sense of meaning for me. If `p` stands for `pairs` or `parameters`, `ps` would be better as well.

Reply:
Probably `platformdata`, since that is what is passed to it.